### PR TITLE
LINKTA-133: Implement footer component

### PR DESCRIPTION
**Pull request by philzmintmojito**
## Description

- Add footer component 
- Update layoutConfig - Added tab objects and `routeToFooterLinks' to map routes to corresponding footer links

Fixes # 133

## Type of change

- [x]  🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨  New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Manual testing by running npm run dev:frontend

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have reviewed the [**Security Best Practices Document**](SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

**Review by kinskiwu — COMMENTED** [review]
Great job on implementing the Footer component! It's well-structured and functional. A few suggestions for refinement:

### General Best Practices:
- **Frequent Commits:** Just FYI, committing more frequently with a more focused scope will make it easier to understand and review changes.

- **Stick to the Ticket:** Ensure that changes are specific to the ticket you're working on. Styling has a separate ticket and should be part of its PR.

- **Collaboration and Coordination**: It's important to coordinate with team members on shared components and files. As Andrew is currently working on routing, which might involve modifications to `main.jsx` and `App.tsx`, let's hold off on making any changes that could conflict with his work. 

- **Testing Insight**:  If you've conducted manual testing by temporarily inserting the Footer into the App, please mention it in your PR submission. 

### PR Specific Notes:

- **Accessibility**:  Good initiative with the footer links. For better accessibility and clarity, ensure each <Link> has a `to` attribute leading to specific destinations. Use placeholders with a // TODO: comment for yet-to-be-determined links.

- **Modifications of Other Components**: I observed some changes to the `App` component. While it's totally ok to temporarily insert your component in `App` for testing, please ensure these changes are reverted before finalizing your PR. 

- **Styling Considerations**: Currently, there are some styling dependencies and code related to Radix themes included in your PR. Since styling is addressed in a separate ticket, please remove any Radix theme packages and associated styling from your PR. If styling interests you, you're welcome to pick up the styling ticket and create a dedicated PR for it.

File: client/App.jsx
Comment (danielboisselle) [inline client/App.jsx:9]: You should put the h1 and div inside the Footer component, keep App.jsx clean with component imports

typically you root file/layout files follow

```ts
const Layout = () => {
  return (
    <div>
      <Header />
      <MainContent />
      <Footer />
    </div>
  );
};
```

Comment (philzmintmojito) [reply client/App.jsx]: Thank you for the help, Daniel. 
I've removed my footer part from the App.jsx.

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (kinskiwu) [inline client/components/layout/main-layout/footer/Footer.tsx:1]: Since we have not decided on using radix ui themes as our styling library and the themes package is not in package.json, please remove `Flex`, `Text`, and `Link` from the component.

For now pls use either use primitive components from radix ui or regular tags, and `Link` from react-router for routing. 

Before you commit new changes, remember to manually test your code (you may do so after the react-router PR is merged so you don't need to set up temp routers). Thank you!

File: client/main.jsx
Comment (kinskiwu) [inline client/main.jsx:1]: Please revert the main.jsx back to its original form as follows to avoid merge conflicts:

```typescript
import React from 'react';
import ReactDOM from 'react-dom/client';
import App from './App';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
```

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (ChTiSh) [inline client/components/layout/main-layout/footer/Footer.tsx:11]: Please consider the pros and cons with ui/li in this case: https://stackoverflow.com/questions/13416866/why-should-i-use-ul-li-in-my-html

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (ChTiSh) [inline client/components/layout/main-layout/footer/Footer.tsx:7]: Please refer to figma file for correct spelling/formatting

**Review by ChTiSh — CHANGES_REQUESTED** [review]
Please kindly check typos and spellings for the footer component. Please use semantic html and leave styling class for the styling ticket.

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (philzmintmojito) [reply client/components/layout/main-layout/footer/Footer.tsx:7]: Updated based on your comments. Besides that, I checked several websites and think the first character should be © (copy right), not @.  Could you revisit the figma file to fix it?

Comment (ChTiSh) [reply client/components/layout/main-layout/footer/Footer.tsx]: Also along with the latest commit, I'm still seeing typos. "Cookie Preferences" is a set term
@philzmintmojito thank you for pointing it out. I will go change on the figma now.
We definitely need more help with reviewing our mockup designs, but please also kindly note that all symbols are for mockups and many are just placeholders for faster churn-outs at the moment. To avoid confusion, would greatly appreciate all figma comments can be left directly on the figma file/file/element in the future.

**Review by dcpavel — APPROVED** [review]
This looks good to me. Thanks for the updates.

**Comment by ChTiSh** [comment]
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/footer

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (philzmintmojito) [reply client/components/layout/main-layout/footer/Footer.tsx]: Thanks, Chen. Typo is fixed.

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (kinskiwu) [inline client/components/layout/main-layout/footer/Footer.tsx:9]: I think you don't need a `<div>` here since you already have a `<ul>`.

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (kinskiwu) [inline client/components/layout/main-layout/footer/Footer.tsx:4]: Remember to add `React.FC`

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (kinskiwu) [inline client/components/layout/main-layout/footer/Footer.tsx:1]: To keep to code DRY, consider using a data structure such as an array or object to store the footer links and map the `<li/>` components or better maintainability. Check out this [article](https://www.dhiwise.com/post/react-lists-and-keys-the-key-to-efficient-rendering). If you opt for dynamic rendering, ensure the key prop is unique and stable (Check out this [article](https://medium.com/coding-at-dawn/what-is-the-react-key-property-why-you-need-it-5eba88c429f5)).

Comment (kinskiwu) [reply client/components/layout/main-layout/footer/Footer.tsx:1]: There is a `layoutConfig` file where you can store the footer links. Just fyi.

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (kinskiwu) [inline client/components/layout/main-layout/footer/Footer.tsx:12]: Please read my comment about how to set the key property.

File: client/components/layout/main-layout/footer/Footer.tsx
Comment (kinskiwu) [inline client/components/layout/main-layout/footer/Footer.tsx:4]: Pls move this comment to the layout config file.

File: client/components/layout/main-layout/layoutConfig.ts
Comment (kinskiwu) [inline client/components/layout/main-layout/layoutConfig.ts:17]: Consider keeping the naming of your variables consistent with the rest of the document.

**Comment by kinskiwu** [comment]
Since you've updated your PR, pls also update the PR description. Also as mentioned before pls mention what tests you have performed to make sure the feature runs as expected.

**Review by kinskiwu — APPROVED** [review]

**Review by ChTiSh — APPROVED** [review]

**Pull request merged**
